### ISSUE TITLE
Persist and restore State machine with multiple state machines by DefaultStateMachinePersister

### ISSUE THREAD

**Issue by LukasMasek** [spring-projects/spring-statemachine#332]
Hello Everyone,
I have found problem with restoring state machine. If machine with multiple sub state machine has been persisted, then during restore operation extended context variables are not propagated into not active sub state machine. 
Master state machine states and transitions:
INIT -> ADMIN_SM <-> USER_SM -> CLEAN -> INIT

States ADMIN_SM and USER_SM represents state with child state machines. 
In case when machine is persisted in state connected with USER_SM, context is stored and what I can say is stored correctly. However when I restore state machine context into different state machine extended context of ADMIN_SM is not updated and doesn't contains map with variables required by my application to be able to work.

coming from AbstractStateMachine implementation on line 699. Where is condition for context propagation into state machine is based on boolean flag which can be true just in case when current state is active state of state machine.
More specifically I'm talking about this peace of code:
		if (stateSet && stateMachineContext.getExtendedState() != null) {
			this.extendedState = stateMachineContext.getExtendedState();
		}

Thank you for response,
BR,
Lukas Masek

PS: To make clear my knowledge of Spring SM are limited. I started with this tool last month. That means I have spent just month and half by working with this library.

**Comment by jvalkeal** [comment]
Yes that looks a bit weird, thx for pointing it out. Think I need to revisit how extended states work with submachines as currently we just try to use extended state from root machine. Ultimate goal has been that submachine could be fully independent which then raises a little bit of awkwardness how machine(s) should be used and accessed.
